src/API/Logs/LogRecord.php

```diff
@@ -42,9 +42,9 @@ public function setContext(?ContextInterface $context = null): self
     /**
      * @see https://opentelemetry.io/docs/reference/specification/logs/data-model/#field-severitynumber
      */
-    public function setSeverityNumber(int $severityNumber): self
+    public function setSeverityNumber(Severity $severityNumber): self
```

Comment:
Since we've removed `emitEvent(LogRecord)`, there shouldn't be any direct usage of this class outside of log appenders (the monolog one in contrib is the only one I'm aware of). So, I don't think this is a critical BC to maintain...?

Reply:
It has very likely low/no impact (as `LogRecord::setSeverityNumber(Psr::serverityNumber($...))` stays compatible); but it is also quite trivial to keep BC. IMO we should preserve BC for the low chance that someone is using the logs API w/ a non-PSR-3 logger.

Reply:
Fair enough, reverted to maintain BC. I also deprecated `Psr::severityNumber` in favour of moving that logic into the enum itself.